WcaOnRails/lib/advancement_condition.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+class AdvancementCondition
+  include ActiveModel::Validations
+
+  attr_accessor :level
+  validates :level, numericality: { only_integer: true }
+
+  def initialize(level)
+    self.level = level
+  end
+
+  def to_wcif
+    { "type" => self.class.wcif_type, "level" => self.level }
+  end
+
+  def ==(other)
+    other.class == self.class && other.to_wcif == self.to_wcif
+  end
+
+  def hash
+    self.to_wcif.hash
+  end
+
+  def self.wcif_type_to_class
+    @@wcif_type_to_class ||= AdvancementCondition.subclasses.map { |cls| [cls.wcif_type, cls] }.to_h
```

Comment:
@larspetrus, let's continue the discussion about this now. The question was how to go from `wcif_type` to "specific subclass of AdvancementCondition" and vice versa. I think you were in favor of explicitly listing the classes rather than looping over the subclasses of `AdvancementCondition`, but I don't see the benefit of that over what we have here.